lib/task/import/validate/csvColumnNameValidator.class.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * CSV column name test. Validate column names against list of valid AtoM import fields.
+ * Output warning if unknown column names found. Output list of unknown column names.
+ * Validate against files in 'lib/flatfile/config'.
+ *
+ * @author     Steve Breker <sbreker@artefactual.com>
+ *
+ * @internal
+ * @coversNothing
+ */
+class CsvColumnNameValidator extends CsvBaseValidator
+{
+    const TITLE = 'Column Name Validation';
+    // Do not reset in between multiple CSVs.
+    protected $validColumnNames = [];
+    protected $validColumnNamesLowercase = [];
+
+    protected $unknownColumnNames = [];
+    protected $caseIssuesColumnNameMap = [];
+    protected $trimIssuesColumnNames = [];
+    protected $complete = false;
+
+    public function __construct(array $options = null)
+    {
+        $this->setTitle(self::TITLE);
+        parent::__construct($options);
+
+        $this->loadObjectColumnNames($this->options['className']);
+    }
+
+    public function reset()
+    {
+        $this->complete = false;
+        $this->unknownColumnNames = [];
+        $this->caseIssuesColumnNameMap = [];
+        $this->trimIssuesColumnNames = [];
+
+        parent::reset();
+    }
+
+    public function testRow(array $header, array $row)
+    {
+        parent::testRow($header, $row);
+
+        // Only do this check once per file.
+        if (!$this->complete) {
+            foreach ($header as $columnName) {
+                // If $columnName is not in validColumnNames.
+                if (!in_array($columnName, $this->validColumnNames)) {
```

Comment:
Opinion: I think this would be easier to read if the valid column check is refactored into a separate method.

Also, again I would prefer:

```
if (!in_array($columnName, $this->validColumnNames)) {
     continue;
}
```

To make it clear that no further processing should be done in the branch, and avoid unnecessary indenting.

Reply:
Both suggestions implemented.